src/main/java/bio/terra/janitor/db/TrackRequest.java

```diff
@@ -0,0 +1,41 @@
+package bio.terra.janitor.db;
+
+import bio.terra.janitor.generated.model.CloudResourceUid;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
+import java.time.Instant;
+import java.util.Map;
+
+/** A request to track a resource. This is a precursor to having a {@link TrackedResource}. */
+@AutoValue
+public abstract class TrackRequest {
```

Comment:
Those AutoValue classes under this directory is really confusing to me. I was trying to understand the use of each classes recently when I worked on RbsDao, then lose track. 
Initially we have this: https://github.com/DataBiosphere/terra-resource-janitor/blob/master/src/main/java/bio/terra/janitor/db/TrackedResource.java 
And I think that make sense to have a java class mapped to db schema object in db layer, and better to differentiate with API layer objects. By then I thought there will be only `number of db tables` java object. But then for each db query result we add a new object class, that make sense as well because it is clearer/simpler to return one single object for a `db retrieve method`. But now, there are so many similar objects, and I feel they are just random combination of `CloudResourceUid, State, creation, trackedResourceId`.  I don't think that is easier for people to understand the usage/difference of each of them. Is there anything we can do about this?

Reply:
We talked some offline about this. The `AndLabels` & `andFlight` TrackedResource classes are a little unfortunate, but there's not another great way to make Pai<TrackedResource, Thing> in Java, and we don't want to have Optional<Thing> that are sometimes not set.
TrackRequest is outside of what's represented in the database as it's a precursor. We don't have a better idea at the moment.